restart_plugin/mtcp_restart_plugin.c

```diff
@@ -594,21 +602,28 @@ mtcp_plugin_hook(RestoreInfo *rinfo)
     //         mtcp_restart is statically linked, and doesn't need it.
     Area heap_area;
     MTCP_ASSERT(getMappedArea(&heap_area, "[heap]") == 1);
-    start1 = MAX(heap_area.endAddr, (VA)rinfo->pluginInfo.memRange.end);
+    start1 = MAX(heap_area.endAddr, (VA)lh_info.memRange.end);
     Area stack_area;
     MTCP_ASSERT(getMappedArea(&stack_area, "[stack]") == 1);
     end1 = MIN(stack_area.endAddr - 4 * GB, rinfo->minHighMemStart - 4 * GB);
-    // Reserve 8MB above min high memory region. That should include space for
-    // stack, argv, env, auxvec.
+    // maxHighMemEnd reserves 8MB above min high memory region.
+    // That should include space for stack, argv, env, auxvec.
     start2 = rinfo->minHighMemStart;
-    end2 = rinfo->minHighMemStart + 8 * MB;
+    end2 = rinfo->maxHighMemEnd;
     // Ignore region start2:end2 if it is overlapped with region start1:end1
     if (is_overlap(start1, end1, start2, end2)) {
       start2 = 0;
       end2 = 0;
     }
   }
-  // FIXME:  End of '#if 1'; Remove '# else' branch when the code is stable.
+
+  Area vvar_area;
+  MTCP_ASSERT(getMappedArea(&vvar_area, "[vvar]") == 1);
```

Comment:
@gc00

Let's either improve this code to call getMappedArea for vvar only if vvar actually exists in the proc maps, or else enclose it in # if ... endif by checking kernel version. This way a single codebase can work on both Perlmutter and the discovery cluster.